addons/binding/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/discovery/OpenUVDiscoveryService.java

```diff
@@ -0,0 +1,116 @@
+public class OpenUVDiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(OpenUVDiscoveryService.class);
+
+    private static final int DISCOVER_TIMEOUT_SECONDS = 10;
+    private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
+
+    private final LocationProvider locationProvider;
+    private final OpenUVBridgeHandler bridgeHandler;
+    private @Nullable ScheduledFuture<?> discoveryJob;
+    private @Nullable PointType previousLocation;
+
+    /**
+     * Creates a OpenUVDiscoveryService with enabled autostart.
+     */
+    public OpenUVDiscoveryService(OpenUVBridgeHandler bridgeHandler, LocationProvider locationProvider) {
+        super(SUPPORTED_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS, true);
+        this.locationProvider = locationProvider;
+        this.bridgeHandler = bridgeHandler;
+    }
+
+    @Override
+    protected void activate(@Nullable Map<String, @Nullable Object> configProperties) {
+        super.activate(configProperties);
+    }
+
+    @Override
+    @Modified
+    protected void modified(@Nullable Map<String, @Nullable Object> configProperties) {
+        super.modified(configProperties);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting OpenUV discovery scan");
+        PointType location = locationProvider.getLocation();
+        if (location == null) {
+            logger.debug("LocationProvider.getLocation() is not set -> Will not provide any discovery results");
+            return;
+        }
+        createResults(location);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (discoveryJob == null) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(() -> {
+                PointType currentLocation = locationProvider.getLocation();
+                if (currentLocation != null && !Objects.equals(currentLocation, previousLocation)) {
+                    logger.debug("Location has been changed from {} to {}: Creating new discovery results",
+                            previousLocation, currentLocation);
+                    createResults(currentLocation);
+                    previousLocation = currentLocation;
+                }
+            }, 0, LOCATION_CHANGED_CHECK_INTERVAL, TimeUnit.SECONDS);
+            logger.debug("Scheduled OpenUV-changed job every {} seconds", LOCATION_CHANGED_CHECK_INTERVAL);
+        }
+    }
+
+    public void createResults(PointType location) {
+        ThingUID localOpenUVThing = new ThingUID(LOCATION_REPORT_THING_TYPE, LOCAL);
+        ThingUID bridgeUID = bridgeHandler.getThing().getUID();
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(LOCATION, location.toString());
+        thingDiscovered(DiscoveryResultBuilder.create(localOpenUVThing).withLabel("Local UV Information")
+                .withProperties(properties).withRepresentationProperty(location.toString()).withBridge(bridgeUID)
```

Comment:
Thanks for all the updates @clinique ! :+1: 

When using a representation property the name of the property is added an not the value. It's also defined in the thing type XML. 

See for instance how it's added in [this Sonos PR](https://github.com/eclipse/smarthome/pull/4035/files) and also the [ESH Discovery Services documentation](https://www.eclipse.org/smarthome/documentation/development/bindings/discovery-services.html). Another example would be the Astro Binding which uses a very similar "geolocation" as representation property.